File: libgdx_galaxy_invaders/core/src/com/nopalsoft/invaders/game_objects/Boost.java
package com.nopalsoft.invaders.game_objects;

public class Boost extends DynamicGameObject {

    public static final float DRAW_SIZE = 5;
    public static final float RADIUS = 1f;
    public static final float SPEED = -10f;

    public static final int EXTRA_LIFE_BOOST = 0;
    public static final int EXTRA_MISSILE_BOOST = 1;
    public static final int EXTRA_SHIELD_BOOST = 2;
    public static final int SHIELD = 3;
    public final int type;
    int stateTime;

    public Boost(int type, float x, float y) {
        super(x, y, RADIUS);
        this.type = type;
        velocity.add(0, SPEED);
        stateTime = 0;
    }

    public void update(float deltaTime) {
        position.add(velocity.x * deltaTime, velocity.y * deltaTime);
        boundsCircle.x = position.x;
        boundsCircle.y = position.y;
        stateTime += deltaTime;
    }
}

File: libgdx_galaxy_invaders/core/src/com/nopalsoft/invaders/game_objects/Boost.kt
package com.nopalsoft.invaders.game_objects

class Boost(@JvmField val type: Int, x: Float, y: Float) : DynamicGameObject(x, y, RADIUS) {
    var stateTime: Int

    init {
        velocity.add(0f, SPEED)
        stateTime = 0
    }

    fun update(deltaTime: Float) {
        position.add(velocity.x * deltaTime, velocity.y * deltaTime)
        boundsCircle!!.x = position.x
        boundsCircle!!.y = position.y
        stateTime = (stateTime + deltaTime).toInt()
    }

    companion object {
        const val DRAW_SIZE: Float = 5f
        const val RADIUS: Float = 1f
        const val SPEED: Float = -10f

        const val EXTRA_LIFE_BOOST: Int = 0
        const val EXTRA_MISSILE_BOOST: Int = 1
        const val EXTRA_SHIELD_BOOST: Int = 2
        const val SHIELD: Int = 3
    }
}
